File: sources/Android/android-communications/library/src/main/java/com/polar/androidcommunications/api/ble/model/gatt/client/pmd/PmdControlPointCommand.kt
package com.polar.androidcommunications.api.ble.model.gatt.client.pmd

enum class PmdControlPointCommandClientToService(val code: Int) {
    NULL_ITEM(0),
    GET_MEASUREMENT_SETTINGS(1),
    REQUEST_MEASUREMENT_START(2),
    STOP_MEASUREMENT(3),
    GET_SDK_MODE_MEASUREMENT_SETTINGS(4),
    GET_MEASUREMENT_STATUS(5),
    GET_SDK_MODE_STATUS(6),
    GET_OFFLINE_RECORDING_TRIGGER_STATUS(7),
    SET_OFFLINE_RECORDING_TRIGGER_MODE(8),
    SET_OFFLINE_RECORDING_TRIGGER_SETTINGS(9);
}

enum class PmdControlPointCommandServiceToClient(val code: Int) {
    ONLINE_MEASUREMENT_STOPPED(1);

    companion object {
        fun fromByte(cpByte: Byte): PmdControlPointCommandServiceToClient? {
            val byteValue = cpByte.toInt() and 0xFF
            return PmdControlPointCommandServiceToClient.values().firstOrNull { it.code == byteValue }
        }
    }
}

File: sources/iOS/ios-communications/Sources/iOSCommunications/ble/api/model/gatt/client/pmd/PmdControlPointCommand.swift
//  Copyright © 2022 Polar. All rights reserved.

import Foundation

struct PmdControlPointCommandClientToService {
    static let GET_MEASUREMENT_SETTINGS: UInt8 = 0x01
    static let REQUEST_MEASUREMENT_START: UInt8 = 0x02
    static let STOP_MEASUREMENT: UInt8 = 0x03
    static let GET_SDK_MODE_SETTINGS: UInt8 = 0x04
    static let GET_MEASUREMENT_STATUS: UInt8 = 0x05
    static let GET_SDK_MODE_STATUS: UInt8 = 0x06
    static let GET_OFFLINE_RECORDING_TRIGGER_STATUS: UInt8 = 0x07
    static let SET_OFFLINE_RECORDING_TRIGGER_MODE: UInt8 = 0x08
    static let SET_OFFLINE_RECORDING_TRIGGER_SETTINGS: UInt8 = 0x09
}

struct PmdControlPointCommandServiceToClient {
    static let ONLINE_MEASUREMENT_STOPPED: UInt8 = 0x01
}
